eth_abi/grammar.py

```diff
@@ -263,12 +263,17 @@ def validate(self):
         base, sub = self.base, self.sub
 
         # Check validity of string type
-        if base in ('string', 'bytes'):
+        if base == 'string':
+            if sub is not None:
+                self.invalidate('string type cannot have suffix')
+
+        # Check validity of bytes type
+        if base == 'bytes':
```

Comment:
Probably `elif` here, for consistent style.

Reply:
Whoops!  Thanks for catching that.